packages/webpack/src/plugins/nx-webpack-plugin/lib/normalize-options.ts

```diff
@@ -76,7 +76,7 @@ export function normalizeOptions(
 
   const sourceRoot = projectNode.data.sourceRoot ?? projectNode.data.root;
 
-  if (!options.main) {
+  if (!combinedPluginAndMaybeExecutorOptions.main) {
```

Comment:
This should be checking the combined options (executor and config) and not only what has been set inside the config